_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -813,147 +813,160 @@ mfxStatus MFXVideoENCODE_EncodeFrameAsync(mfxSession session, mfxEncodeCtrl *ctr
-            (MFX_ERR_MORE_BITSTREAM == mfxRes))
-        {
-            // prepare the obsolete kind of task.
-            // it is obsolete and must be removed.
-            if (NULL == entryPoints[0].pRoutine)
-            {
-                MFX_TASK task;
-
-                memset(&task, 0, sizeof(task));
-                // BEGIN OF OBSOLETE PART
-                task.bObsoleteTask = true;
-                task.obsolete_params.encode.internal_params = internal_params;
-                // fill task info
-                task.pOwner = session->m_pENCODE.get();
-                task.entryPoint.pRoutine = &MFXVideoENCODELegacyRoutine;
-                task.entryPoint.pState = session->m_pENCODE.get();
-                task.entryPoint.requiredNumThreads = 1;
-
-                // fill legacy parameters
-                task.obsolete_params.encode.ctrl = ctrl;
-                task.obsolete_params.encode.surface = reordered_surface;
-                task.obsolete_params.encode.bs = bs;
-                // END OF OBSOLETE PART
-
-                task.priority = session->m_priority;
-                task.threadingPolicy = session->m_pENCODE->GetThreadingPolicy();
-                // fill dependencies
-                task.pSrc[0] = surface;
-                task.pDst[0] = ((mfxStatus)MFX_ERR_MORE_DATA_SUBMIT_TASK == mfxRes) ? 0: bs;
-
-                task.pSrc[1] = bs;
-                task.pSrc[2] = ctrl ? ctrl->ExtParam : 0;
-
-#ifdef MFX_TRACE_ENABLE
-                task.nParentId = MFX_AUTO_TRACE_GETID();
-                task.nTaskId = MFX::CreateUniqId() + MFX_TRACE_ID_ENCODE;
-#endif // MFX_TRACE_ENABLE
-
-                // register input and call the task
-                MFX_CHECK_STS(session->m_pScheduler->AddTask(task, &syncPoint));
-            }
-            else if (1 == numEntryPoints)
+        try
+        {
+            mfxSyncPoint syncPoint = NULL;
+            mfxFrameSurface1 *reordered_surface = NULL;
+            mfxEncodeInternalParams internal_params;
+            MFX_ENTRY_POINT entryPoints[MFX_NUM_ENTRY_POINTS];
+            mfxU32 numEntryPoints = MFX_NUM_ENTRY_POINTS;
+
+            memset(&entryPoints, 0, sizeof(entryPoints));
+            mfxRes = session->m_pENCODE->EncodeFrameCheck(ctrl,
+                                                        surface,
+                                                        bs,
+                                                        &reordered_surface,
+                                                        &internal_params,
+                                                        entryPoints,
+                                                        numEntryPoints);
+            // source data is OK, go forward
+            if ((MFX_ERR_NONE == mfxRes) ||
+                (MFX_WRN_INCOMPATIBLE_VIDEO_PARAM == mfxRes) ||
+                (MFX_WRN_OUT_OF_RANGE == mfxRes) ||
+                // WHAT IS IT??? IT SHOULD BE REMOVED
```

Comment:
maybe, let's fix this one?)
1) remove comment
2) fix the code
3) write something more meaningful instead

Reply:
definitely not in this commit..